source/Nevermore.IntegrationTests/Advanced/UseExistingTransactionFixture.cs

```diff
@@ -0,0 +1,138 @@
+
+        using (new AssertionScope())
+        {
+            const string expectedMessage = "An existing connection and transaction were provided, they should have been opened externally";
+
+            await ((Func<Task>)(async () => await writeTransaction.OpenAsync(cancellationToken)))
+                .Should()
+                .ThrowExactlyAsync<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+
+            ((Action)(() => writeTransaction.Open()))
+                .Should()
+                .ThrowExactly<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+
+            await ((Func<Task>)(async () => await writeTransaction.OpenAsync(IsolationLevel.ReadCommitted, cancellationToken)))
+                .Should()
+                .ThrowExactlyAsync<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+
+            ((Action)(() => writeTransaction.Open(IsolationLevel.ReadCommitted)))
+                .Should()
+                .ThrowExactly<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+        }
+    }
+
+    [Test]
+    public async Task CommittingNonOwnedTransaction_Throws()
+    {
+        await using var sqlConnection = new SqlConnection(ConnectionString);
+        await sqlConnection.OpenAsync(cancellationToken);
+        await using var sqlTransaction = sqlConnection.BeginTransaction();
+
+        var nonOwnedTransaction = Store.CreateWriteTransactionFromExistingConnectionAndTransaction(sqlConnection, sqlTransaction);
+
+        using (new AssertionScope())
+        {
+            const string expectedMessage = $"{nameof(WriteTransaction)} cannot commit a transaction it does not own";
+
+            await ((Func<Task>)(async () => await nonOwnedTransaction.CommitAsync(cancellationToken)))
+                .Should()
+                .ThrowExactlyAsync<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+
+            ((Action)(() => nonOwnedTransaction.Commit()))
+                .Should()
+                .ThrowExactly<InvalidOperationException>()
+                .WithMessage(expectedMessage);
+        }
+    }
+
+    [Test]
+    public async Task DisposingNonOwnedTransaction_DoesNotDisposeConnectionOrTransaction()
+    {
+        await using var sqlConnection = new SqlConnection(ConnectionString);
+        await sqlConnection.OpenAsync(cancellationToken);
+        await using var sqlTransaction = sqlConnection.BeginTransaction();
+
+        var nonOwnedTransaction = Store.CreateWriteTransactionFromExistingConnectionAndTransaction(sqlConnection, sqlTransaction);
+        nonOwnedTransaction.Dispose();
+
+        using (new AssertionScope())
+        {
+            // ReSharper disable once AccessToDisposedClosure
+            ((Action)(() => sqlTransaction.Commit())).Should().NotThrow();
+            sqlConnection.State.Should().Be(ConnectionState.Open);
+        }
+    }
```

Comment:
Some good edge cases covered 👍 

Would it be possible to write a test case for the happy path for an externally owned connection? 

e.g. start a non-owned transaction, do some work, commit, verify changes?